libr/core/cmd_search.c

```diff
@@ -908,14 +908,276 @@ static RList* construct_rop_gadget(RCore *core, ut64 addr, ut8 *buf, int idx,
 	return hitlist;
 }
 
+static RList* parse_list (char *str) {
+	RList *list;
+	char *line, *data, *p;
+
+	if (!str) {
+		return NULL;
+	}
+
+	list = r_list_newf (free);
+	line = strtok (strdup (str), "\n");
+	data = strchr ( line, '=');
+	p = strtok (data + 1, ",");
+
+	while (p) {
+		r_list_append (list, (void*)strdup (p));
+		p = strtok (NULL, ",");
+	}
+
+	return list;
+}
+
+static RList* get_constants (char *str) {
+	RList *list;
+	char *p, *data;
+
+	if (!str) {
+		return NULL;
+	}
+
+	data = strdup (str);
+	list = r_list_newf (free);
+	p = strtok (data, ",");
+
+	while (p) {
+		if (strtol (p, NULL, 0)) {
+			r_list_append (list, (void*)strdup (p));
+		}
+		p = strtok (NULL, ",");
+	}
+
+	return list;
+}
+
+
+static char* rop_classify_constant (RCore *core, RList *ropList) {
+	char *esil_str, *constant;
+	char *mov = NULL;
+	RListIter *iter_src, *iter_r, *iter_l, *iter_dst, *iter_const;
+	RRegItem *item_src, *item_dst;
+	RList *head, *constants;
+	RList *ops_list = NULL, *flg_read = NULL, *flg_write = NULL, *reg_read = NULL,
+		*reg_write = NULL, *mem_read = NULL, *mem_write = NULL;
+	int romem = r_config_get_i (core->config, "esil.romem");
+	int stats = r_config_get_i (core->config, "esil.stats");
+
+	if (!romem || !stats) {
+		// eprintf ("Error: esil.romem and esil.stats must be set TRUE");
+		return NULL;
+	}
+
+	r_list_foreach (ropList, iter_r, esil_str) {
+		if (strchr (esil_str, '[')) { // avoid MEM read/write for now
+			return NULL;
+		}
+
+		constants = get_constants (esil_str);
+		// if there are no constants in the instruction continue
+		if (!constants || !constants->head) {
+			continue;
+		}
+
+		// init regs with random values
+		head = r_reg_get_list (core->dbg->reg, 0);
+		if (!head) {
+			return NULL;
+		}
+		r_list_foreach (head, iter_dst, item_dst) {
+			int r = r_num_rand (30000);
```

Comment:
im not a big fan of random behaviours. this can turn into unexpected results on each run. i would choose to use predefined values before entering the loop. so it doesnt slowdowns the emulation and you can just specify valid values pointing to available memory